apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/store.ts

```diff
@@ -21,12 +21,10 @@ export const selectors = {
 
 const STORE_KEY = 'automattic/starter-page-layouts';
 
-registerStore( STORE_KEY, {
-	// In reality the store can dispatch any action, however `reducer` has a
-	// strongly typed action type to make the typings inside the function
-	// easier to work with.
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	reducer: reducer as any,
-	actions,
-	selectors,
-} );
+export const store = createReduxStore( STORE_KEY, { reducer, actions, selectors } );
+
+// TODO: Why is this needed?
+if ( ! starterPageTemplatesConfig.didRegisterStore ) {
+	register( store );
+	starterPageTemplatesConfig.didRegisterStore = true;
+}
```

Comment:
@jsnajdr I'm curious if you have any thoughts about the best way to avoid registering the store twice.

From what I can tell, the `@wordpress/data` package overwrites the existing store every time you call `register`: https://github.com/WordPress/gutenberg/blob/605aeb0f4f7d2225120e498f95ae27b9f56d77a3/packages/data/src/registry.js#L217-L265

(It doesn't check if the store name was already registered, and just writes it to the stores object under the store's unique key).

This isn't normally an issue, if a file is only executed once when importing it.

But what if there are multiple different bundles trying to access the same store? In ETK, each sub-feature is its own webpack bundle. So starter page templates has its own bundle which is conditionally loaded when applicable, and a different feature like "wpcom block editor nux" has a different bundle which is conditionally loaded when needed. (When a bundle is needed depends on different factors like what page you visit.)

So we end up with this:

```js
// file1, which is in one bundle:
import { store } from './store';

// file2, which is in a different bundle
import { store as templateStore } from '../../starter-page-templates/store';
```

When the bundle associated with file1 is loaded, it sets up some initial store based on some conditionals: https://github.com/Automattic/wp-calypso/blob/2656f2beeca506ac256b71be878ac05f42b17832/apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/index.tsx#L29-L44

But when the bundle associated with file2 is loaded, register is called again, which overwrites the store which clears out that change.

As far as I can tell, after this point, dispatch/select will be using the same store from any bundle. But it'd be nice to avoid clearing out the existing store the second time it gets registered!

I'm wondering if there's a better way to do this other than relying on globals, because I think we'll want similar behavior for some other stores.